druid.yaml

```diff
@@ -0,0 +1,85 @@
+package:
+  name: druid
+  version: 30.0.0
+  epoch: 0
+  description: Apache Druid is a high performance real-time analytics database.
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - bash
+      - busybox
+      - openjdk-17-default-jvm
+
+environment:
+  contents:
+    packages:
+      - bash
+      - busybox
+      - maven
+      - openjdk-17
+      - openjdk-17-default-jvm
+      - py3-pyyaml
+      - python3
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/druid.git
+      tag: druid-${{package.version}}
+      expected-commit: 09d36ee324747f1407705c27618b6d415c3fa8a9
+      cherry-picks: |
+        master/75937c98e82156d8812143a047baf3f2a8af6b72: CVE-2023-3635
+
+  - runs: |
+      mvn -B -ff -q \
+      install \
+      -Pdist,bundle-contrib-exts \
+      -Pskip-static-checks,skip-tests \
+      -Dmaven.javadoc.skip=true -T1C
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/share/java/druid
+      tar -zxf distribution/target/apache-druid-${{package.version}}-bin.tar.gz --strip-components=1 -C ${{targets.destdir}}/usr/share/java/druid
+
+  - runs: |
+      # deduplicate jars
+      cp distribution/docker/deduplicate_jars.sh ${{targets.destdir}}/deduplicate_jars.sh
+      ${{targets.destdir}}/deduplicate_jars.sh ${{targets.destdir}}/usr/share/java/druid
+
+subpackages:
+  - name: "${{package.name}}-compat"
+    description: "Compatibility package to copy the entrypoint scripts"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"
+          cp distribution/docker/druid.sh ${{targets.subpkgdir}}/druid.sh
+          cp distribution/docker/peon.sh ${{targets.subpkgdir}}/peon.sh
+
+          sed -i "s|/opt/druid|/usr/share/java/druid|" ${{targets.subpkgdir}}/druid.sh
+          sed -i "s|/opt/druid|/usr/share/java/druid|" ${{targets.subpkgdir}}/peon.sh
+    dependencies:
+      runtime:
+        - iproute2
+
+test:
+  environment:
+    contents:
+      packages:
+        - python3
+        - perl
+  pipeline:
+    - runs: |
+        /usr/share/java/druid/bin/start-nano-quickstart > /usr/share/java/druid/logs.txt 2>&1 &
+        DRUID_PID=$!
+        sleep 2
+        grep -q "Starting Apache Druid" /usr/share/java/druid/logs.txt
+        kill $DRUID_PID
```

Comment:
It seems Druid [provides an API](https://druid.apache.org/docs/latest/api-reference/service-status-api/) to service status, should we do quick health check using `curl` just after `grep` tests?

FWIW, you can even use `test/daemon-check-output` here.

Reply:
The entrypoint script that lets you use the API needs a cluster to work correctly. I have included it in the image PR though: https://github.com/chainguard-images/images-private/blob/f00a7b2f6b5382cffc5189f68024cb83b81d3760/images/druid/tests/test.sh#L12

Thanks for the tip about `test/daemon-check-output`, will look into that